lib/shopify_api/resources/assigned_fulfillment_order.rb

```diff
@@ -0,0 +1,12 @@
+module ShopifyAPI
+  class AssignedFulfillmentOrder < Base
+    CANCELLATION_REQUESTED = 'cancellation_requested'
+    FULFILLMENT_REQUESTED = 'fulfillment_requested'
+    FULFILLMENT_ACCEPTED = 'fulfillment_accepted'
+
+    def self.all(options = {})
+      assigned_fulfillment_orders = super(options)
```

Comment:
We could go either way. If we don't perform that check on this side there's not really any use for the constants described above.

From an API perspective I would argue that all checks that you can perform without making a request across the network should be performed locally. 

One could argue that doing so kind of forces you to keep these constant values in sync with Core but with versioning and the 1 year deprecation cycle you should have plenty of time to update them if anything changes on Core.

Reply:
That's fair, I'm fine with both implementations 👍

Reply:
My original thought was to provide the constants to prevent typos from custom client code. Putting the constants/checks here means that it should be kept in sync with server side changes. The rest of the shopify_api doesn't seem to define many constants, so removing them is more consistent.

Reply:
I think the `assigned Fulfillment orders` controller is probably the only one that accepts a custom type of filter param (assignment status) that's later converted into something else. I think that's the reason why you can't find another example on the `shopify api`.

I'm fine either way as I previously said but you should keep in mind that relying on Shopify Core to return you a `bad request` when you send in invalid `assignment status` means that you will have to parse that error message appropriately in order to return something meaningful to the user.

I ran a quick test and this is the error message Core will return :

`"Assignment status is not included in the list"`

This conveys what the issue is but isn't clear from an API standpoint IMO (what's the list). One way you could address this, could be through the documentation. Just something to keep in mind IMO.

Reply:
That's not a great error message, but if the client specifies an `assigned_status` and got this Bad Request, it should be enough to point to documentation.